libminifi/src/core/Processor.cpp

```diff
@@ -277,6 +286,28 @@ void Processor::onTrigger(const std::shared_ptr<ProcessContext> &context, const
   }
 }
 
+void Processor::callOnTrigger(const std::shared_ptr<ProcessContext>& context, const std::shared_ptr<ProcessSession>& session) {
+  uint64_t num_incoming_flow_files_queued_before = numIncomingFlowFilesQueued();
+  onTrigger(context, session);  // virtual method overridden by subclasses
+  uint64_t num_incoming_flow_files_queued_after = numIncomingFlowFilesQueued();
+
+  if (num_incoming_flow_files_queued_before != 0 && num_incoming_flow_files_queued_before == num_incoming_flow_files_queued_after) {
+    logger_->log_info("Processor %s processed no incoming flow files (out of %" PRIu64 "), so it will yield", name_, num_incoming_flow_files_queued_before);
+    yield();
+  }
+}
+
+void Processor::callOnTrigger(ProcessContext* context, ProcessSession* session) {
```

Comment:
Is this different from the other overload? Can the other one not just call to this one?

Reply:
They call different `onTrigger()` overloads.  But I will need a different approach in any case, so this is moot.